### ISSUE TITLE
Regression in base@2.13.1 with list options in object previews

### ISSUE THREAD

**Issue by geoffreyjball** [sanity-io/sanity#2631]
**Describe the bug**

Starting in @sanity/base@2.13.1, list options (on a `type: string`) are not implemented in a preview when using the `prepare()` function. The bug does not occur if the `prepare()` function is omitted. It occurs on both object and document previews.

**To Reproduce**

1. Create a string field that is configured with `options.list` values:
```javascript
{
  title: "Size",
  name: "size",
  type: "string",
  options: {
    list: [
      { title: "Small", value: "small" },
      { title: "Medium", value: "medium" },
      { title: "Large", value: "large" },
      { title: "X-Large", value: "xlarge" },
      { title: "Full", value: "full" },
    ],
  },
},
```
2. Create a `preview` that includes a `prepare` function:
```javascript
preview: {
  select: {
    size: "size",
  },
  prepare(selection) {
    const { size } = selection;
    return {
      title: size ? `Size: ${size}` : null;
    };
  },
},
```
3. Run `sanity start` and create a document/object.
4. Preview will return a blank title. In base@2.13.0, the title is returned correctly. Commenting out `options` in the schema will also correctly return the values in the preview.

**Expected behavior**

It's expected that the title will be returned correctly (i.e., displaying `Size: <size>`) in the preview.

**Which versions of Sanity are you using?**

base@2.13.1

**What operating system are you using?**

macOS 11.3

**Which versions of Node.js / npm are you running?**

7.18.1
v16.4.1

**Additional context**

The regression appears related to the code that was added [here](https://github.com/sanity-io/sanity/commit/c1d9d8fee3ddbc7b96b85fea98a645996e270720#diff-8b30eb59a556413ef80f0f6061d599f22016e644008caff7d715732a12edb42c), as commenting out the resulting code in `node_modules/@sanity/base/lib/preview/prepareForPreview.js` causes the preview to work as expected. That code is:
```javascript
    // Find the field the value belongs to
    var valueField = type === null || type === void 0 ? void 0 : (_type$fields = type.fields) === null || _type$fields === void 0 ? void 0 : _type$fields.find(f => f.name === selection[key]); // Check if predefined options exist

    var listOptions = valueField === null || valueField === void 0 ? void 0 : (_valueField$type = valueField.type) === null || _valueField$type === void 0 ? void 0 : (_valueField$type$opti = _valueField$type.options) === null || _valueField$type$opti === void 0 ? void 0 : _valueField$type$opti.list;

    if (listOptions) {
      // Find the selected option that matches the raw value
      var selectedOption = listOptions.find(opt => opt.value === rawValue[selection[key]]);
      acc[key] = (0, _get2.default)(selectedOption, key);
      return acc;
    }
```

**Comment by vicbergquist** [comment]
I noticed this too, looking into it!

**Comment by rudfoss** [comment]
Just adding that I managed to reproduce the problem without `prepare`. Here is my complete schema file:

```typescript
// First, we must import the schema creator
import createSchema from 'part:@sanity/base/schema-creator'

// Then import schema types from any plugins that might expose them
import schemaTypes from 'all:part:@sanity/base/schema-type'

const obj = {
	title: "Obj",
	name: "obj",
	type: "object",
	fields: [
		{
			title: "Plain text",
			name: "plain",
			type: "string"
		},
		{
			title: "Pick something",
			name: "pickSomething",
			type: "string",
			options: {
				list: [
					{ title: "Alpha", value: "a" },
					{ title: "Bravo", value: "b" },
					{ title: "Charlie", value: "c" },
					{ title: "Delta", value: "d" },
					{ title: "Echo", value: "e" },
					{ title: "Foxtrot", value: "f" },
				]
			}
		}
	],
	preview: {
		select: {
			title: "plain",
			subtitle: "pickSomething"
		}
	}
}

const doc = {
	title: "Doc",
	name: "doc",
	type: "document",
	fields: [
		{
			title: "Pickers",
			name: "pickers",
			type: "array",
			of: [{ type: "obj" }]
		}
	]
}

// Then we give our schema to the builder and provide the result to Sanity
export default createSchema({
  // We name our schema
  name: 'default',
  // Then proceed to concatenate our document type
  // to the ones provided by any plugins that are installed
  types: schemaTypes.concat([
    /* Your types here! */
    obj,
    doc
  ]),
})
```

**Comment by KittyGiraudel** [comment]
I was just about to open an issue about this—I’m glad to see I’m not the only one who stumbled into this. If that helps, a workaround is to add to every option from the field you’re previewing a key matching the one you query in `select`.

Considering @geoffreyjball’s original example:

```js
{
  title: "Size",
  name: "size",
  type: "string",
  options: {
    list: [
      { title: "Small", value: "small", size: "small" },
      { title: "Medium", value: "medium", size: "medium" },
      { title: "Large", value: "large", size: "large" },
      { title: "X-Large", value: "xlarge", size: "xlarge" },
      { title: "Full", value: "full", size: "full" },
    ],
  },
},
```

Not ideal, but also very actionable until there is a fix.

**Comment by dsite** [comment]
Hi team! I'm relieved to see that there's a fix for this. Is there an ETA on when it will be released? I've been bitten by this bug, and I've seen four or five posts on Slack in the last few days from people with the same issue.

**Comment by paulgalow** [comment]
Just a quick note: Upgrading to [base@2.14.0](https://github.com/sanity-io/sanity/releases/tag/v2.14.0) has (unexpectedly) solved my preview issue.

**Comment by geoffreyjball** [comment]
Resolved in v2.14.0.
